good-enough-practices-for-scientific-computing.tex

```diff
@@ -1284,26 +1341,77 @@ \subsection*{Supplementary Materials}
 computational work (which increasingly means \emph{all} work) more
 reproducible.
 
-But progress will not happen by itself.  Universities and funding
-agencies need to support training for researchers in the use of these
-tools.  Such investment will improve confidence in the results of
+But progress will not happen by itself. The practices are being 
+increasingly incentivized through requirements from funding
+agencies and journals, but the time and skills required to put these
+approaches into practice are still not being valued. 
+
+At a local level,
+PIs can have the most impact, requiring that the research their lab
+produces follow these recommendations. Even if a PI doesn't have
+a background in computation, they can require that students show
+and share their code in lab meetings and with lab mates, that data
+is available and accessible to all in the lab and that computational
+methods sections are comprehensive. Best practices can include
+things like paired-programming and having other lab members replicate
+workflows.\ref{computing_workflows_for_biologists}. PIs can also value the time
+it takes to do these things effectively and provide opportunities for
+training. While it seems challenging to take time away from other
+research activities, like coding itself, the time spent on the initial efforts
+will pay off with more efficient and likely more innovative research.
+
+Universities can provide support for advice and infrastructure. This is
+most commonly provided by IT or High Performance Computing groups, 
+and an often under appreciated resource, the libraries. Librarians have
+thought about and worked with data and provenance even before 
+these computational challenges and increasingly
+have dedicated data librarians on staff who have an explicit service role.
+Many campuses also have self-organizing student groups, as students
+seek opportunities to engage with and learn from each other around 
+these skills. Universities would do well to support these groups, as 
+student peers are often some of the most knowledgable and they 
+help grow communities of practice. Universities can also provide 
+opportunities for training on these practical skills. Many resources are
+available already. They would do well to make use of existing training
+programs and resources, and even contribute to them, as the training 
+challenges are shared by all universities, and there is not a need to replicate 
+efforts. Additionally, where courses are not yet available, they can provide
+access to short courses and workshops, where again materials and often 
+instructors are already provided, so they can more quickly provide training. 
+In our experience with Software and Data Carpentry, our two-day workshops
+at universities are almost always full within a day or two of being announced,
+with wait lists as long or longer than the number of attendees. It is clear that there
+is extensive demand for training and unmet needs. 
```

Comment:
While I agree with everything here, this paragraph has drifted into a more editorial tone, by making recommendations for university administrations, for example.